src/common/commonutils/UrlUtils.c

```diff
@@ -3,24 +3,26 @@
 
 #include "Internal.h"
 
-char* UrlEncode(char* target)
+char* UrlEncode(const char* target, size_t targetSize)

```

Comment:
Do we need special handling for a newline character?

Reply:
See line 30